blob/blob.go

```diff
@@ -76,22 +90,12 @@ func NewBlob(_ uint8, nID namespace.ID, data []byte) (*Blob, error) {
 	return &Blob{Blob: *blob, commitment: com}, nil
 }
 
-// NamespaceID returns blobs namespaceId.
-func (b *Blob) NamespaceID() []byte {
-	return b.Blob.NamespaceId
-}
-
-// Data returns blobs raw data.
-func (b *Blob) Data() []byte {
-	return b.Blob.Data
+// Namespace returns blob's namespace.
+func (b *Blob) Namespace() namespace.ID {
+	return append([]byte{uint8(b.NamespaceVersion)}, b.NamespaceId...)
 }
 
 // Commitment returns the commitment of current blob.
 func (b *Blob) Commitment() Commitment {
 	return b.commitment
 }
```

Comment:
On second thought, we can just make the commitment Public. There is no need to encapsulate it as we always compute it with the constructor. We don't compute it when we interact with the Blob